src/components/Card.js

```diff
@@ -1,60 +1,79 @@
-export default class Card {
-    constructor({ name, link }, cardSelector, handleImageClick) {
-      this._name = name;
-      this._link = link;
-      this._cardSelector = cardSelector;
-      this._handleImageClick = handleImageClick;
+class Card {
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  ) {
+    this._name = data.name;
+    this._link = data.link;
+    this._id = data._id;
+    this._isLiked = data.isLiked;
+    this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
+    this._cardSelector = cardSelector;
+  }
+
+  _getTemplate() {
+    const cardElement = document
+      .querySelector(this._cardSelector)
+      .content.firstElementChild.cloneNode(true);
+
+    return cardElement;
+  }
+
+  isLiked() {
+    return this._isLiked;
+  }
+
+  handleLike = () => {
```

Comment:
Needs correction: please remove this method, you don't need it anymore. Instead you should create a method for updating isLiked status and use it after successful api response in index.js:

```js
setIsLiked(isLiked) {
  this._isLiked = isLiked;
  this.updateLikeView();
}
```

Reply:
Like this?

Reply:
Yes, use that method in index.js after likes update request in `then` block for updating the likes status and the button state

Reply:
function handleLikeClick(card) {
  api
    .updateLike(card._id, card.isLiked())
    .then(() => {
      card.handleLike();
      console.log(card);
      setIsLiked(isLiked) 
        this._isLiked = isLiked;
        this.updateLikeView();
      
})
    .catch(console.error);
}

Reply:
like this?